src/Filesystem/FilesystemWalker.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace LustreCollector.Filesystem;
+
+public class FilesystemWalker
+{
+    public static async Task Walk(string mountPoint, CancellationToken stoppingToken, Action<LustreFile> processor)
+    {
+        var roots = new Stack<string>();
+        roots.Push(mountPoint);
+
+        while (roots.Count > 0)
+        {
+            var root = roots.Pop();
+            string[] childRoots;
+            string[] childFiles;
+
+            try
+            {
+                childRoots = Directory.GetDirectories(root);
+                childFiles = Directory.GetFiles(root);
+            }
+            catch (FileNotFoundException ex)
+            {
+                // Root has been removed from under us while processing.
+                continue;
+            }
+            
+            foreach (string file in childFiles)
+            {
+                if (stoppingToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                try {
+                    var info = new FileInfo(file);
+                    if (info.LinkTarget != null)
+                    {
+                        processor(new LustreFile(root, info.Name, info.Length,
+                            info.LastAccessTime.ToFileTimeUtc()));
```

Comment:
The DLCS Orchestrator is calling `File.SetLastWriteTime` last write time when checking if a file exists - is that the correct attr to change? Or should it be `File.SetLastAccessTime` ?